posthog/temporal/batch_exports/s3_batch_export.py

```diff
@@ -503,6 +503,8 @@ async def flush_to_s3(last_uploaded_part_timestamp: str, last=False):
 
             await s3_upload.complete()
 
+        return local_results_file.records_total
```

Comment:
Also de-dented, but it competes with the `s3_upload` which happens at an outer scope...

Maybe I'm being un-Pythonic? The use of important information outside of the `with` gives me the creeps, but maybe this is fine and good. I guess it's fine as long as `__exit__` leaves the state we need?

Reply:
`await s3_upload.complete()` should be in the `__exit__` of `S3MultiPartUpload`. I took it out as I was going crazy trying to debug a batch export completing with an extra number of parts, so I tried being explicit to throw some clarity at the problem.

In the end, I think I randomly fixed the bug in another PR when I wasn't looking for it. Anyways, this remained outside of the `__exit__` and should be re-added in as it wasn't the cause of the bug.

Reply:
When we move it we do have to account for exceptions: We don't want to complete the upload in case of an exception and we have to be very careful if we are aborting. I can open a follow-up PR to deal with this.